File: CovidTransmission.java
/** 
 * Copyright (c) 2020 @rappelrx on GitHub
 */

import java.util.Scanner;

/**
 * This class uses the Scanner class to determine total 
 * time of contact based on user input in the format:
 * (startDay startHour startMin endDay endHour endMin).
 */
public class CovidTransmission {
    
    // Magic numbers
    private static final int MAX_DAY = 31;
    private static final int MAX_HOUR = 23;
    private static final int MAX_MIN = 59;
    private static final int MINS_IN_DAY = 1440;
    private static final int SEC_IN_MIN = 60;
    private static final int HRS_IN_DAY = 24;
    private static final int THREE_HOURS = 180;
    private static final int SIX_HOURS = 360;
    private static final String EMPTY_STRING = "";
    private static final String LOW_RISK = "low";
    private static final String MED_RISK = "medium";
    private static final String HIGH_RISK = "high";
    private static final String EXT_HIGH_RISK = "extremely high";

    /** 
     * Prints total time of contact (in minutes) and
     * risk level of contracting Covid based on this 
     * total time of contact.
     * 
     * @param args, not used
     * @return void, nothing returned
     */
    public static void main(String[] args) {
        Scanner scnr = new Scanner(System.in);
        int numMinutes = 0; // initialize
        String riskLevel = EMPTY_STRING; // initialize

        String d1Input = scnr.next();
        String h1Input = scnr.next();
        String m1Input = scnr.next();
        String d2Input = scnr.next();
        String h2Input = scnr.next();
        String m2Input = scnr.next();

        int d1 = Integer.parseInt(d1Input);
        int h1 = Integer.parseInt(h1Input);
        int m1 = Integer.parseInt(m1Input);
        int d2 = Integer.parseInt(d2Input);
        int h2 = Integer.parseInt(h2Input);
        int m2 = Integer.parseInt(m2Input);

        // Check for invalid inputs.
        if ((d1 < 0 || d1 > MAX_DAY) || (d2 < 0 || d2 > MAX_DAY)) {
            numMinutes = -1;
            riskLevel = "-1";
        }
        if ((h1 < 0 || h1 > MAX_HOUR) || (h2 < 0 || h2 > MAX_HOUR)) {
            numMinutes = -1;
            riskLevel = "-1";
        }
        if ((m1 < 0 || m1 > MAX_MIN) || (m2 < 0 || m2 > MAX_MIN)) {
            numMinutes = -1;
            riskLevel = "-1";
        }

        // Determine number of minutes of contact.
        if (riskLevel != "-1") {
            numMinutes = (d2 - d1) * MINS_IN_DAY;

            if (h2 < h1) {
                numMinutes -= MINS_IN_DAY;
                numMinutes += ((HRS_IN_DAY - h1 + h2) * SEC_IN_MIN);
            } else {
                numMinutes += (h2 - h1) * SEC_IN_MIN;
            }

            if (m2 < m1) {
                numMinutes -= SEC_IN_MIN;
                numMinutes += (SEC_IN_MIN - m1 + m2);
            } else {
                numMinutes += (m2 - m1);
            }
        }

        // Determine risk level of contracting Covid.
        if (numMinutes >= 0 && numMinutes <= SEC_IN_MIN) {
            riskLevel = LOW_RISK;
        } else if (numMinutes > SEC_IN_MIN && numMinutes <= THREE_HOURS) {
            riskLevel = MED_RISK;
        } else if (numMinutes > THREE_HOURS && numMinutes <= SIX_HOURS) {
            riskLevel = HIGH_RISK;
        } else if (numMinutes > SIX_HOURS) {
            riskLevel = EXT_HIGH_RISK;
        }

        // Invalid number of minutes.
        if (numMinutes < 0) {
            numMinutes = -1;
            riskLevel = "-1";
        }

        System.out.println(numMinutes + " " + riskLevel);
        scnr.close();
    }
}

File: CovidTransmission.cpp
/** 
 * Copyright (c) 2023 @rappelrx on GitHub
 *
 * This program determines total time of contact based on terminal input,
 * and determines the risk level of contracting Covid.
 * 
 * Disclaimer:
 * The information used in this program is not fact-checked.
 * Please refer to CDC [cdc.gov] for facts about Covid.
 */

#include <iostream> 
#include <cstdio> // scanf is faster than cin
#include <string>
using namespace std;

/** 
 * This struct contains the constant variables that represent
 * the various risk levels of contracting Covid, ranging from
 * "low" to "extremely high."
 */
struct CovidTransmission {
    const string EMPTY_STRING = "";
    const string LOW_RISK = "low";
    const string MED_RISK = "medium";
    const string HIGH_RISK = "high";
    const string EXT_HIGH_RISK = "extremely high";
};

// Global constants replace magic numbers
const unsigned int g_max_day = 31;
const unsigned int g_max_hour = 23;
const unsigned int g_max_min = 59;
const unsigned int g_mins_in_day = 1440;
const unsigned int g_sec_in_min = 60;
const unsigned int g_hrs_in_day = 24;
const unsigned int g_three_hours = 180;
const unsigned int g_six_hours = 360;

/**
 * Helper method for checking validity of input for day.
 * 
 * @param input The int value representation for day.
 * @return true if input is within bounds, otherwise false.
 */
bool isValidDay(int input) {
    if (input < 0 || input > g_max_day) 
        return false;
    return true;
}

/**
 * Helper method for checking validity of input for hour.
 * 
 * @param input The int value representation for hour.
 * @return true if input is within bounds, otherwise false.
 */
bool isValidHour(int input) {
    if (input < 0 || input > g_max_hour) 
        return false;
    return true;
}

/**
 * Helper method for checking validity of input for minute.
 * 
 * @param input The int value representation for minute.
 * @return true if input is within bounds, otherwise false.
 */
bool isValidMinute(int input) {
    if (input < 0 || input > g_max_min)
        return false;
    return true;
}

int main() {
    int d1, h1, m1, d2, h2, m2;
    cout << "Please enter initial day, hour, min: ";
    scanf("%d %d %d", &d1, &h1, &m1);
    cout << "Please enter ending day, hour, min: ";
    scanf("%d %d %d", &d2, &h2, &m2);

    // Check validity of inputs.
    if (!isValidDay(d1) || !isValidDay(d2)) {
        cerr << "Invalid input for day." << endl;
        return 1;
    }
    if (!isValidHour(h1) || !isValidHour(h2)) {
        cerr << "Invalid input for hour." << endl;
        return 1;
    }
    if (!isValidMinute(m1) || !isValidMinute(m2)) {
        cerr << "Invalid input for minute." << endl;
        return 1;
    }

    // Define struct to access variables for risk levels.
    struct CovidTransmission covidRisks;
    string riskLevel = covidRisks.EMPTY_STRING;

    // Determine number of minutes of contact.
    int numMinutes = (d2 - d1) * g_mins_in_day;
    if (h2 < h1) {
        numMinutes -= g_mins_in_day;
        numMinutes += (g_hrs_in_day - h1 + h2) * g_sec_in_min;
    } else {
        numMinutes += (h2 - h1) * g_sec_in_min;
    }
    if (m2 < m1) {
        numMinutes -= g_sec_in_min;
        numMinutes += g_sec_in_min - m1 + m2;
    } else {
        numMinutes += m2 - m1;
    }

    // Determine risk level of contracting Covid.
    if (numMinutes >= 0 && numMinutes <= g_sec_in_min) {
        riskLevel = covidRisks.LOW_RISK;
    } else if (numMinutes > g_sec_in_min && numMinutes <= g_three_hours) {
        riskLevel = covidRisks.MED_RISK;
    } else if (numMinutes > g_three_hours && numMinutes <= g_six_hours) {
        riskLevel = covidRisks.HIGH_RISK;
    } else if (numMinutes > g_six_hours) {
        riskLevel = covidRisks.EXT_HIGH_RISK;
    }

    // Invalid number of minutes.
    if (numMinutes < 0) {
        cerr << "Number of minutes cannot be negative." << endl;
        return 1;
    }

    cout << numMinutes << " minutes of contact puts you at "
    << riskLevel << " risk of contracting Covid." << endl;

    return 0;
}